src/sentry/backup/comparators.py

```diff
@@ -759,7 +759,8 @@ def get_default_comparators():
         list,
         {
             "sentry.apitoken": [
-                HashObfuscatingComparator("refresh_token", "token", "token_last_characters"),
+                HashObfuscatingComparator("refresh_token", "token"),
+                IgnoredComparator("token_last_characters"),
```

Comment:
The only thing this is used for is showing the user a hint as to what token we're talking about in the UI? If so, I think this is fine.

If there is some load-bearing security verification that relies on these last 4 chars, we may want to write a custom comparator to check them. An easy option is to just "flip" the `EqualOrRemovedComparator`, which basically checks that the old value is equal to the new one, or else the new value is null. So in this case, we'd have an `EqualOrAddedComparator` that checks if the values are equal, or else the old value is null.

Reply:
Yeah, this column isn't used for any security verification. Just a quality-of-life thing used in the UI when tokens are displayed in an obfuscated way. Right now that logic is happening entirely in the view/serializer code. 

Ultimately, we will have to figure these out on imports once tokens a fully hashed. We'll need to decide if we're going to import user API tokens, regenerate them, or omit them.

Reply:
Ok, sgtm.

> Ultimately, we will have to figure these out on imports once tokens a fully hashed. We'll need to decide if we're going to import user API tokens, regenerate them, or omit them.

Atm we honor them in all cases, meaning you can import your API tokens from your self-hosted instance and they will keep working. I suppose this leaves some room for users to shoot themselves in the foot (manually set an insecure token on their self-hosted instance, then blame us when it causes problems after being imported into SaaS), but that doesn't seem very likely. Flip side is that it's a lot easier to tell people "just point your API calls at this new domain, and don't worry about the Bearer token".

In any case, that discussion is out scope for this PR.  Approved :)

Reply:
Wait, on further reflection: we only import these for self-hosted users anyway, see relocation scopes [here](https://github.com/getsentry/sentry/blob/98c2b184e9740dc7ed348fb4db177733ae6e6b0f/src/sentry/models/apitoken.py#L33) (we only use `RelocationScope.Organization` for SaaS imports). So it seems like you can't take your ApiTokens with you to SaaS in any case.